File: Server/src/main/content/global/skill/production/crafting/item/PotteryCraftPulse.java
package content.global.skill.production.crafting.item;

import content.global.skill.production.crafting.data.PotteryData;
import core.game.node.entity.player.Player;
import core.game.node.entity.player.link.diary.DiaryType;
import core.game.node.entity.skill.SkillPulse;
import core.game.node.entity.skill.Skills;
import core.game.node.item.Item;
import core.game.world.map.Location;
import core.game.world.update.flag.context.Animation;
import core.utilities.StringUtils;

import static core.api.ContentAPIKt.setAttribute;

public final class PotteryCraftPulse extends SkillPulse<Item> {

    private static final Animation ANIMATION = Animation.create(896);
    private static final Item SOFT_CLAY = new Item(1761);
    private final PotteryData pottery;
    private int amount;
    private int ticks;

    public PotteryCraftPulse(Player player, Item node, int amount, final PotteryData pottery) {
        super(player, node);
        this.pottery = pottery;
        this.amount = amount;
    }

    @Override
    public boolean checkRequirements() {
        if (!player.getInventory().contains(1761, 1)) {
            player.getPacketDispatch().sendMessage("You need soft clay in order to do this.");
            return false;
        }
        if (player.getSkills().getLevel(Skills.CRAFTING) < pottery.getLevel()) {
            player.getPacketDispatch().sendMessage("You need a crafting level of " + pottery.getLevel() + " to make this.");
            return false;
        }
        return true;
    }

    @Override
    public void animate() {
        if (ticks % 5 == 0) {
            player.animate(ANIMATION);
        }
    }

    @Override
    public boolean reward() {
        if (++ticks % 5 != 0) {
            return false;
        }
        if (player.getInventory().remove(SOFT_CLAY)) {
            if (pottery == PotteryData.BOWL && player.getLocation().withinDistance(Location.create(3086, 3410, 0))) {
                setAttribute(player, "/save:diary:varrock:spun-bowl", true);
            }
            final Item item = pottery.getUnfinished();
            player.getInventory().add(item);
            player.getSkills().addExperience(Skills.CRAFTING, pottery.getExp(), true);
            player.getPacketDispatch().sendMessage("You make the soft clay into " + (StringUtils.isPlusN(pottery.getUnfinished().getName()) ? "an" : "a") + " " + pottery.getUnfinished().getName().toLowerCase() + ".");
            if (pottery == PotteryData.POT && player.getLocation().withinDistance(Location.create(3086, 3410, 0))) {
                player.getAchievementDiaryManager().finishTask(player, DiaryType.LUMBRIDGE, 0, 7);
            }
        }
        amount--;
        return amount < 1;
    }

}

File: Server/src/main/content/global/skill/production/crafting/item/PotteryCraftPulse.kt
package content.global.skill.production.crafting.item

import content.global.skill.production.crafting.data.PotteryData
import core.api.*
import core.api.consts.Animations
import core.api.consts.Items
import core.game.node.entity.player.Player
import core.game.node.entity.player.link.diary.DiaryType
import core.game.node.entity.skill.SkillPulse
import core.game.node.entity.skill.Skills
import core.game.node.item.Item
import core.game.world.map.Location
import core.game.world.update.flag.context.Animation
import core.tools.StringUtils

/**
 * Pottery craft pulse
 *
 * @property amount
 * @property pottery
 * @constructor
 *
 * @param player
 * @param node
 */
class PotteryCraftPulse(player: Player?, node: Item?, var amount: Int, val pottery: PotteryData) : SkillPulse<Item?>(player, node) {

    var ticks = 0

    override fun checkRequirements(): Boolean {
        if (getStatLevel(player, Skills.CRAFTING) < pottery.level) {
            sendMessage(player, "You need a crafting level of " + pottery.level + " to make this.")
            return false
        }
        if (!inInventory(player, Items.SOFT_CLAY_1761, 1)) {
            sendMessage(player, "You need soft clay in order to do this.")
            return false
        }
        return true
    }

    override fun animate() {
        if (ticks % 5 == 0) {
            animate(player, ANIMATION)
        }
    }

    override fun reward(): Boolean {
        if (++ticks % 5 != 0) {
            return false
        }
        if (player.inventory.remove(SOFT_CLAY)) {
            if (pottery == PotteryData.BOWL && withinDistance(player, Location(3086, 3410, 0))) {
                setAttribute(player, "/save:diary:varrock:spun-bowl", true)
            }
            val item = pottery.unfinished
            player.inventory.add(item)
            rewardXP(player, Skills.CRAFTING, pottery.exp)
            sendMessage(player, "You make the soft clay into " + (if (StringUtils.isPlusN(pottery.unfinished.name)) "an" else "a") + " " + pottery.unfinished.name.lowercase() + ".")

            if (pottery == PotteryData.POT && withinDistance(player, Location(3086, 3410, 0))) {
                finishDiaryTask(player, DiaryType.LUMBRIDGE, 0, 7)
            }
        }
        amount--
        return amount < 1
    }

    companion object {
        private val ANIMATION = Animation.create(Animations.HUMAN_COOKING_RANGE_896)
        private val SOFT_CLAY = Item(Items.SOFT_CLAY_1761)
    }
}
